linera-core/src/remote_node.rs

```diff
@@ -194,20 +197,25 @@ impl<N: ValidatorNode> RemoteNode<N> {
             }
         };
 
-        let mut missing_blobs = blob_ids;
+        let mut missing_blob_ids = blob_ids.into_iter().collect::<HashSet<_>>();
         let mut found_blobs = if let Some(info) = info {
-            let new_found_blobs = missing_blobs
+            let new_found_blobs = info
+                .manager
+                .locked_pending_blobs
                 .iter()
-                .filter_map(|blob_id| info.manager.pending_blobs.get(blob_id))
+                .filter(|blob| missing_blob_ids.contains(&blob.id()))
                 .map(|blob| (blob.id(), blob.clone()))
                 .collect::<HashMap<_, _>>();
-            missing_blobs.retain(|blob_id| !new_found_blobs.contains_key(blob_id));
+            missing_blob_ids.retain(|blob_id| !new_found_blobs.contains_key(blob_id));
```

Comment:
It feels like you should be able to do the above in one go (one loop over `info.manager.locked_pending_blobs`).